Find a functionally equivalent code:com.android.keyguard.KeyguardFaceUnlockView.maybeStartBiometricUnlock:COMMENT
Method Modifier: private     
Comment:/**
 * Starts the biometric unlock if it should be started based on a number of factors.  If it
 * should not be started, it either goes to the back up, or remains showing to prepare for
 * it being started later.
 */

@@ -4,16 +4,15 @@
     if (mBiometricUnlock != null) {
         KeyguardUpdateMonitor monitor = KeyguardUpdateMonitor.getInstance(mContext);
         final boolean backupIsTimedOut = (monitor.getFailedUnlockAttempts() >= LockPatternUtils.FAILED_ATTEMPTS_BEFORE_TIMEOUT);
-        PowerManager powerManager = (PowerManager) mContext.getSystemService(Context.POWER_SERVICE);
-        boolean isShowing;
-        synchronized (mIsShowingLock) {
-            isShowing = mIsShowing;
+        boolean isBouncerVisibleToUser;
+        synchronized (mIsBouncerVisibleToUserLock) {
+            isBouncerVisibleToUser = mIsBouncerVisibleToUser;
         }
-        // showing.
-        if (!powerManager.isScreenOn() || !isShowing) {
+        // it here and ready for when the bouncer does show.
+        if (!isBouncerVisibleToUser) {
             // It shouldn't be running but calling this can't hurt.
             mBiometricUnlock.stop();
             return;

